File: Misc DMOJ/binary/src/binary.java
import java.io.*;
import java.util.*;
public class binary {
    /*
     * Copy-pasting code is NOT cool! Please do not copy and paste my code as a submission to DMOJ.
     * github.com/jdabtieu/competitive-programming
     */
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer in;
    
    public static void main(String[] args) throws IOException {
        for (int cases = readInt(); cases > 0; cases--) {
            String n = Integer.toBinaryString(readInt());
            while (n.length() % 4 != 0) n = "0" + n;
            for (int i = 0; i < n.length(); i += 4) {
                System.out.print(n.substring(i, i + 4) + " ");
            }
            System.out.println();
        }
    }
    
    static String next() throws IOException {
        while (in == null || !in.hasMoreTokens())
            in = new StringTokenizer(br.readLine());
        return in.nextToken();
    }
    
    static long readLong() throws IOException {
        return Long.parseLong(next());
    }
    
    static int readInt() throws IOException {
        return Integer.parseInt(next());
    }
    
    static double readDouble() throws IOException {
        return Double.parseDouble(next());
    }
    
    static char readChar() throws IOException {
        return next().charAt(0);
    }
    
    static String readLine() throws IOException {
        return br.readLine();
    }
}

File: Misc DMOJ/binary/src/binary.cpp
#include <bits/stdc++.h>

typedef unsigned long long llu;
typedef long long ll;
typedef unsigned int uint;
typedef __uint128_t lllu;
typedef __int128_t lll;

using namespace std;
/*
 * Copy-pasting code is NOT cool! Please do not copy and paste my code as a submission to DMOJ.
 * github.com/jdabtieu/competitive-programming
 */
int main() {
    int cases; scanf("%d", &cases);
    while (cases--) {
        vector<char> v;
        int n; scanf("%d", &n);
        if (n == 0) {
            puts("0000");
            continue;
        }
        while (n > 0) {
            v.insert(v.begin(), (n & 1) + '0');
            n >>= 1;
        }
        int t = 4 - v.size() % 4;
        if (t != 4) while (t--) v.insert(v.begin(), '0');
        for (int i = 0; i < v.size(); i++) {
            printf("%c", v[i]);
            if (i % 4 == 3) printf(" ");
        }
        puts("");
    }
}